client/core/core.go

```diff
@@ -1008,13 +1011,57 @@ func (c *Core) connectedWallet(assetID uint32) (*xcWallet, error) {
 	return wallet, nil
 }
 
+func (c *Core) connectWallet(w *xcWallet) error {
+	err := w.Connect(c.ctx)
+	if err != nil {
+		return newError(walletErr, "error connecting %s wallet: %v", unbip(w.AssetID), err)
+	}
+	// If the wallet is not synced, start a loop to check the sync status until
+	// it is.
+	if !w.synced {
+		// If the wallet is shut down before sync is complete, exit the syncer
+		// loop.
+		innerCtx, cancel := context.WithCancel(c.ctx)
+		go func() {
+			w.connector.Wait()
+			cancel()
+		}()
+		// The syncer loop.
+		go func() {
+			for {
+				ticker := time.NewTicker(syncTickerPeriod)
```

Comment:
I think you want the NewTicker outside the loop otherwise you get a new one each iteratoin.  Also a defer ticker.Stop to prevent a ticker leak.

Reply:
Third try's the charm.